app/domain/authentication/authn_jwt/validate_status.rb

```diff
@@ -0,0 +1,77 @@
+module Authentication
+  module AuthnJwt
+
+    ValidateStatus = CommandClass.new(
+      dependencies: {
+        fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
+        discover_identity_provider: Authentication::OAuth::DiscoverIdentityProvider.new,
+        create_signing_key: Authentication::AuthnJwt::CreateSigningKeyInterface.new,
+        fetch_issuer_value: Authentication::AuthnJwt::FetchIssuerValue.new,
+        validate_uri_parameters: Authentication::AuthnJwt::ValidateUriBasedParameters.new,
+        validate_generic_status_validations: Authentication::AuthnJwt::ValidateGenericStatusValidations.new,
+        fetch_identity_from_token: Authentication::AuthnJwt::IdentityFromDecodedTokenProvider
+      },
+      inputs: %i[authenticator_status_input]
+    ) do
+
+      def call
+        validate_generic_status_validations
+        create_authentication_parameters
+        validate_service_id_exists
+        validate_uri_based_parameters
+        validate_secrets
+      end
+
+      private
+
+      def validate_generic_status_validations
+        @validate_generic_status_validations.call(
+          authenticator_status_input: @authenticator_status_input
+        )
+      end
+
+      def validate_uri_based_parameters
+        @validate_uri_parameters.call(authenticator_input: @authenticator_status_input,
+                                          enabled_authenticators: Authentication::InstalledAuthenticators.enabled_authenticators_str(ENV)
+        )
+      end
+
+      def validate_secrets
+        validate_signing_key_secrets
+        validate_issuer
+        validate_identity_secrets
+      end
+
+      def validate_signing_key_secrets
+        @create_signing_key.call(authentication_parameters: @authentication_parameters)
+      end
+
+      def validate_issuer
+        @fetch_issuer_value.call(authentication_parameters: @authentication_parameters)
+      end
+
+      def validate_identity_secrets
+        @fetch_identity_from_token.new(@authentication_parameters).identity_configured_properly?
+      end
+
+      def required_variable_names
+        @required_variable_names ||= %w[provider-uri]
+      end
+
+      def create_authentication_parameters
+        authentication_input = Authentication::AuthenticatorInput.new(
```

Comment:
if `@authentication_parameters` is initialized you are doing this action for free
You can return the  the `@authentication_parameters` if initialized at the start of the function